network-api/networkapi/wagtailpages/models.py

```diff
@@ -556,13 +558,59 @@ class IndexPage(FoundationMetadataPageMixin, Page):
         FieldPanel('intro'),
     ]
 
+    def filter_entries_for_tag(self, context):
+        """
+        Realise the 'entries' queryset and filter it for tags presences.
+        We need to perform this realisation because there is no guarantee
+        that all children for this IndexPage in fact have a `tags` field,
+        so in order to test this each entry needs to be "cast" into its
+        specific model before we can test for whether i) there are tags
+        to work with and then ii) those tags match the specified ones.
+        """
+        type = self.filtered.get('type')
+        if type == 'tag':
+            terms = self.filtered.get('terms')
+            entries = [
+                entry
+                for
+                entry in context['entries'].specific()
+                if
+                hasattr(entry, 'tags')
+                and not
+                set([str(tag) for tag in entry.tags.all()]).isdisjoint(terms)
+            ]
+            context['entries'] = entries
+
     def get_context(self, request):
+        """
+        Bootstrap this page in similar fashion to the PrimaryPage,
+        but include an `entries` context variable that represents
+        all public children under this page.
+
+        Additionally, if this is a fall-through render due to a
+        tag filtering subroute call, perform that filtering of
+        all entries.
+        """
         context = super().get_context(request)
         context = set_main_site_nav_information(self, context, 'Homepage')
         context = get_page_tree_information(self, context)
         context['entries'] = self.get_children().live().order_by('-first_published_at')
+        if hasattr(self, 'filtered'):
+            self.filter_entries_for_tag(context)
         return context
 
+    @route(r'^tags/(?P<tag>[^/]+)/$')
```

Comment:
Is this from Wagtail? Routable_page.models?

Reply:
Exactly - https://www.accordbox.com/blog/wagtail-tutorials-routable-page/